config/application.rb

```diff
@@ -59,6 +59,22 @@ class Application < Rails::Application
     # Therefore we disable "per_form_csrf_tokens" for the time being.
     config.action_controller.per_form_csrf_tokens = false
 
+    # Enable CORS for API v0
+    # (logging is only activated when debug is enabled)
+    config.middleware.insert_before 0, Rack::Cors, debug: Rails.env.development?, logger: (-> { Rails.logger }) do
+      allow do
+        origins do |source, _env|
+          source # echo back the client's `Origin` header instead of using `*`
+        end
+
+        # allowed public APIs
+        %w[articles comments listings podcast_episodes tags users videos].each do |resource_name|
+          # allow read operations, disallow custom headers (eg. api-key) and disable preflight caching
+          resource "/api/#{resource_name}/*", methods: %i[head get options], headers: [], max_age: -1
```

Comment:
what does `max_age: -1` mean here? No max age?

Reply:
means "don't cache the preflight request", the one with `OPTIONS`. the reason why I disabled it it's because I want to do test all CORS endpoint manually once in production, when I'm sure it will all works I'll enable it again

Reply:
Yo guys, sorry for intervening but how are we supposed to send a request to the API if `api-key` header is blocked?

Reply:
Hi @rsalunga29, I understand the conundrum. We felt it was safe enough for now only to enable CORS requests to public endpoints, we'd like to enable it for all authenticated endpoints eventually but we want to understand in depth potential security concerns and find a way not to unintentionally encourage people to hardcode their API keys in publicly accessible frontends leading to other potential security issues.

In the meantime I'd encourage you to calls to endpoints that need authentication to a server, this is a potential workaround: https://dev.to/jrsofty/cors-problem-workaround-for-consuming-dev-to-api-on-your-website-44jb

Reply:
Hello @rhymes, appreciate the response. I understand the security problems facing the API, I also experienced this before coming across this comment. This seems to be a hard problem to solve without using a backend app.